lightning/src/chain/channelmonitor.rs

```diff
@@ -1282,9 +1282,10 @@ impl<Signer: WriteableEcdsaChannelSigner> ChannelMonitor<Signer> {
 	)
 	where L::Target: Logger
 	{
+		let logger = WithChannelMonitor::from(logger, self);
 		self.inner.lock().unwrap().provide_initial_counterparty_commitment_tx(txid,
```

Comment:
Given `WithChannelMonitor` takes the `inner` lock twice -- and we take the lock again after creating the logger --  should we instead use a `WithChannelMonitorImpl` to have everything under one lock?